modules/config/escalations.tf

```diff
@@ -1,36 +1,42 @@
 resource "opsgenie_escalation" "this" {
-  for_each = module.this.enabled ? { for escalation in local.escalations : escalation.name => escalation } : {}
+  for_each = module.this.enabled ? { for escalation in local.escalations : escalation.name => escalation } : tomap()
 
   name        = each.value.name
   description = try(each.value.description, each.value.name)
 
   # Look up our team id by name
   owner_team_id = try(opsgenie_team.this[each.value.owner_team_name].id, null)
 
-  rules {
-    condition   = try(each.value.rule.condition, "if-not-acked")
-    notify_type = try(each.value.rule.notify_type, "default")
-    delay       = try(each.value.rule.delay, 0)
+  dynamic "rules" {
+    for_each = try(each.value.rules, [])
 
-    dynamic "recipient" {
-      for_each = try(each.value.rule.recipients, [])
-
-      content {
-        type = recipient.value.type
-
-        id = lookup(recipient.value, "id", null) != null ? recipient.value.id : (
-          recipient.value.type == "team" ? opsgenie_team.this[recipient.value.team_name].id : (
-            recipient.value.type == "user" ? try(opsgenie_user.this[recipient.value.user_name].id, data.opsgenie_user.this[recipient.value.user_name].id) : (
-              null
+    content {
+      condition   = try(rules.value.condition, "if-not-acked")
+      notify_type = try(rules.value.notify_type, "default")
+      delay       = try(rules.value.delay, 0)
+
+      dynamic "recipient" {
+        for_each = try(rules.value.recipients, [])
+
+        content {
+          type = recipient.value.type
+
+          id = lookup(recipient.value, "id", null) != null ? recipient.value.id : (
+            recipient.value.type == "team" ? opsgenie_team.this[recipient.value.team_name].id : (
+              recipient.value.type == "user" ? try(opsgenie_user.this[recipient.value.user_name].id, data.opsgenie_user.this[recipient.value.user_name].id) : (
+                recipient.value.type == "schedule" ? opsgenie_schedule.this[recipient.value.schedule_name].id : (
```

Comment:
`recipient.value.schedule_name` - will it work and not throw errors if `recipient.value.type != "schedule"` ?